### ISSUE TITLE
XCFramework support

### ISSUE THREAD

**Issue by hhrvoic** [emqx/CocoaMQTT#464]
When exporting this library as a `.xcframework` users will get errors inside inside `.swiftinterface file` when including it in the project: 

![image](https://user-images.githubusercontent.com/11478044/174537160-5b5135c0-1933-4525-bcef-292c018114cc.png)

The name of the type/class (CocoaMQTT) and the module (CocoaMQTT) are the same so they clash in the `.swiftinterface file`. Compiler thinks that all the types are under CocoaMQTT **class**, not under the CocoaMQTT **module** => limitation where if a module and a type have the same name, as usage is assumed to be the type first. One of the simpler ways of avoiding this is to rename the `CocoaMQTT` class to something else. This would however be the breaking change. 

Similar issue: https://github.com/parse-community/Parse-SDK-iOS-OSX/issues/1601

**Comment by leeway1208** [comment]
I think the quickest way is change  the class name (CocoaMQTT) to (project)CocoaMQTT or something else.

**Comment by leeway1208** [comment]
I think the quickest way is changing the class name from (CocoaMQTT)  to (project)CocoaMQTT or something else.

**Comment by leeway1208** [comment]
I think the quickest way is change the class name from (CocoaMQTT)  to (project)CocoaMQTT or something else.

**Comment by hhrvoic** [comment]
Yeah, not sure about naming conventions on this project so I guess it's anything that works :) 
I see there is a `CocoaMQTT5` class for the v5 of the protocol, maybe the original one which is used for the 3.1.1. could be renamed the `CocoaMQTT3`?

**Comment by hhrvoic** [comment]
This is also now represented as a warning in Xcode. 
https://github.com/apple/swift/issues/56573

<img width="958" alt="Screenshot 2023-07-06 at 15 50 53" src="https://github.com/emqx/CocoaMQTT/assets/11478044/d0961fd9-18d0-4bb4-8e02-62f0b34822b8">


Any plans with this?
